jenkins_jobs/modules/publishers.py

```diff
@@ -102,6 +102,11 @@ def archive(parser, xml_parent, data):
         success = XML.SubElement(archiver, 'onlyIfSuccessful')
         success.text = str(data.get('only-if-success', False)).lower()
 
+    if 'only-successful' in data:
```

Comment:
Current master has ```only-if-success```: https://github.com/openstack-infra/jenkins-job-builder/blob/f2b01f7c5ca29f6cf3fa1040106ed9d81fad32af/jenkins_jobs/modules/publishers.py#L136

Actually the block above it does that too so I prefer to rewrite our jobs and not include this patch.